README.md

```diff
@@ -18,11 +19,23 @@ Ultramodern is a reimplementation of much of the core functionality of libultra.
 
 Platform-specific I/O is handled via callbacks that are provided by the project using ultramodern. This includes reading from controllers and playing back audio samples.
 
-Currently, ultramodern depends directly on [RT64](https://github.com/rt64/rt64) for rendering. It may be decoupled in the future to allow using other renderers for targeting different platforms.
+ultramodern expects the user to provide and regiter a graphics renderer. The recommended one is [RT64](https://github.com/rt64/rt64).
 
-### Librecomp
+## Librecomp
 
 Librecomp is a library meant to be used to bridge the gap between code generated by N64Recomp and ultramodern. It provides wrappers to allow recompiled code to call ultramodern. Librecomp also provides some of the remaining libultra functionality that ultramodern doesn't provide, which includes:
+
 * Overlay handling
 * PI DMA (ROM reads)
-* EEPROM, SRAM and Flashram saving (these may be partially moved to ultramodern in the future)
\ No newline at end of file
+* EEPROM, SRAM and Flashram saving (these may be partially moved to ultramodern in the future)
+
+## Building
+
+It is recommended to reference this project on your `CMakeLists.txt` file. Note that this project has been developed with `clang` 15, older versions may not work.
```

Comment:
This phrasing feels a little clunky to me and should probably have some more info about other compilers, how about:
"The recommended usage of these libraries is to include them in your project's `CMakeLists.txt` file via `add_subdirectory`. This project requires C++20 support and was developed using clang 15, older versions of clang may not work. Recent enough versions of MSVC and gcc should work as well, but are not regularly tested."